spec/datadog/integration_spec.rb

```diff
@@ -42,27 +44,15 @@ def thread_count
     end
 
     context 'for file descriptors' do
-      def open_file_descriptors
-        # Unix-specific way to get the current process' open file descriptors and the files (if any) they correspond to
-        Dir['/dev/fd/*'].each_with_object({}) do |fd, hash|
-          hash[fd] =
-            begin
-              File.realpath(fd)
-            rescue SystemCallError # This can fail due to... reasons, and we only want it for debugging so let's ignore
-              nil
-            end
-        end
-      end
```

Comment:
Moved to a shared spec helper, now that this logic is needed in multiple places.

Reply:
I decided not to use this approach for HTTP testing, but I feel like this refactor is still good.

Reply:
I should move this out of this PR.